src/canvas-extractor/canvas_extractor/__main__.py

```diff
@@ -70,65 +88,58 @@ def parse_args():
     end_date = arguments.end_date
 
 
+@catch_exceptions
 def configure_logging():
     global logger
     global error_tracker
 
     logger = logging.getLogger(__name__)
 
-    level = os.environ.get("LOGLEVEL", "INFO")
+    # We only want to log requests information on DEBUG level
+    if log_level == "INFO":
+        canvasapi_logger = logging.getLogger("canvasapi.requester")
+        canvasapi_logger.setLevel("WARNING")
+
     logging.basicConfig(
         handlers=[
             logging.StreamHandler(sys.stdout),
         ],
         format="%(asctime)s - %(levelname)s - %(name)s - %(message)s",
-        level=level,
+        level=log_level,
     )
     error_tracker = ErrorHandler()
 
 
-def _get_courses(
-    canvas: Canvas, sync_db: sqlalchemy.engine.base.Engine
-) -> Tuple[List[Course], DataFrame]:
+@catch_exceptions
+def _get_courses(canvas: Canvas, sync_db: sqlalchemy.engine.base.Engine) -> None:
     logger.info("Extracting Courses from Canvas API")
     (courses, courses_df) = extract_courses(canvas, start_date, end_date, sync_db)
     # Temporary - just for demonstration until UDM mapping
     courses_df.to_csv("data/courses.csv", index=False)
-    return (courses, courses_df)
+    results_store["courses"] = (courses, courses_df)
 
 
-def _get_sections(
-    courses: List[Course], sync_db: sqlalchemy.engine.base.Engine
-) -> Tuple[List[Section], DataFrame]:
+@catch_exceptions
+def _get_sections(sync_db: sqlalchemy.engine.base.Engine) -> None:
     logger.info("Extracting Sections from Canvas API")
+    (courses, _) = results_store["courses"]
```

Comment:
What's the thinking behind moving to using "results_store"?  I like being able to see all of a function's dependencies its parameter list.

Reply:
In the main function there were some assignments based on the return value from each function, if we stop returning them, we can controll what to do when a specific function fails, you can see the "succeeded" validations in the main method and how the decorator wraps the function and returns true or false depending on the resolution of the Callable.